packages/bvaughn-architecture-demo/src/suspense/createGenericCache.ts

```diff
@@ -9,6 +9,14 @@ export interface GenericCache<TParams extends Array<any>, TValue> {
   getValueSuspense(...args: TParams): TValue;
   getValueAsync(...args: TParams): Promise<TValue>;
   getValueIfCached(...args: TParams): { value: TValue } | undefined;
+  addValue(value: TValue, ...args: TParams): void;
+}
+
+export interface GenericCache2<TExtra, TParams extends Array<any>, TValue> {
```

Comment:
`GenericCache2` doesn't seem like a great name. (The suffix "2" doesn't really provide any hint as to what may be different.)

Can we just update `GenericCache` to make the extra param _optional_ somehow (and only have _one_ of these things?)

Reply:
> Can we just update GenericCache to make the extra param optional somehow (and only have one of these things?)

I'd love to but I don't know how (since you can't have an optional parameter *and* rest parameters).

Reply:
You could change the parameter for a given cache to be an object that has some optional attributes and some required attributes.

Reply:
But then we'd be losing some typing information: for example `getFramesSuspense()` needs a `ReplayClient` but `getFramesIfCached()` doesn't, but we'd have an optional `ReplayClient` attribute for both of these functions instead.

Reply:
Interesting.

I don't love having two APIs that are basically the same but for this detail but we can always iterate more on it in future PRs 😁